lib/wasm/src/environ/dummy.rs

```diff
@@ -159,7 +159,7 @@ impl<'dummy_environment> DummyFuncEnvironment<'dummy_environment> {
     // Create a signature for `sigidx` amended with a `vmctx` argument after the standard wasm
     // arguments.
     fn vmctx_sig(&self, sigidx: SignatureIndex) -> ir::Signature {
-        let mut sig = self.mod_info.signatures[sigidx].clone();
+        let mut sig = self.mod_info.signatures[sigidx.index()].clone();
```

Comment:
What would be the difference between `PrimaryMap` and `SecondaryMap` in this case?

Reply:
Ah, that's a good point. I was thinking about `SecondaryMap` here because the indices are defined by an external source (wasm), however they are defined sequentially, so they'd line up with the indices a `PrimaryMap` would provide, so `PrimaryMap` would work here as well, and be a little simpler (`index`/`index_mut` are faster).

Reply:
Thank you for code reviewing.
You suggested using SecondaryMap, but DummyModuleInfo uses PrimaryMap to represent functions and function_bodies fields. So I also use SecondaryMap to make their structures the same.